unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -25,26 +25,95 @@ export function buildCategoryFacetResults(numberOfResults = 11, numberOfRequeste
 
 export function CategoryFacetTest() {
   describe('CategoryFacet', () => {
+    let options: ICategoryFacetOptions;
     let test: IBasicComponentSetup<CategoryFacet>;
     let simulateQueryData: ISimulateQueryData;
 
     beforeEach(() => {
+      options = { field: '@field' };
       simulateQueryData = buildCategoryFacetResults();
       initializeComponent();
     });
 
     function initializeComponent() {
       test = Mock.advancedComponentSetup<CategoryFacet>(
         CategoryFacet,
-        new Mock.AdvancedComponentSetupOptions(null, { field: '@field' }, env => env.withLiveQueryStateModel())
+        new Mock.AdvancedComponentSetupOptions(null, options, env => env.withLiveQueryStateModel())
       );
       test.cmp.activePath = simulateQueryData.query.categoryFacets[0].path;
+
+      spyOn(test.cmp.logger, 'warn');
     }
 
     function allCategoriesButton() {
       return $$(test.cmp.element).find('.coveo-category-facet-all-categories');
     }
 
+    describe('testing collapse/expand', () => {
+      function validateExpandCollapse(shouldBeCollapsed: boolean) {
+        expect($$(test.cmp.element).hasClass('coveo-dynamic-category-facet-collapsed')).toBe(shouldBeCollapsed);
+      }
+
+      function initializeComponentWithCollapse(enableCollapse: boolean, collapsedByDefault: boolean) {
+        options.enableCollapse = enableCollapse;
+        options.collapsedByDefault = collapsedByDefault;
+        initializeComponent();
+        Simulate.query(test.env, simulateQueryData);
+      }
+
+      it(`when enableCollapse & collapsedByDefault options are true
+      facet should be collapsed`, () => {
+        initializeComponentWithCollapse(true, true);
+        validateExpandCollapse(true);
+      });
+
+      it(`when enableCollapse is false & collapsedByDefault options is true
+      facet should not be collapsed`, () => {
+        initializeComponentWithCollapse(false, true);
+
+        validateExpandCollapse(false);
+      });
+
+      it(`allows to collapse`, () => {
```

Comment:
It would be good to mention `when enableCollapse is true` in this and the next test description,